app/src/main/java/com/duckduckgo/app/browser/autofill/AutofillCredentialsSelectionResultHandler.kt

```diff
@@ -166,6 +173,79 @@ class AutofillCredentialsSelectionResultHandler @Inject constructor(
         return credentialSaver.updateCredentials(originalUrl, selectedCredentials, updateType)
     }
 
+    suspend fun processGeneratePasswordResult(
+        result: Bundle,
+        viewModel: BrowserTabViewModel,
```

Comment:
Overall the methods in this class seemed tightly couple to the view model to the point that perhaps they should be in the view model instead? In this case for example, the fragment calls this method and this method calls the view model so perhaps should be the other way around? Fragment -> ViewModel -> Handler

Reply:
Yeah, it is indeed tightly coupled. I originally (before just this PR) had more logic in the view model but it was already a chunky view model and hard to test parts of it. So I wanted the handler to take a lot of that responsibility away from the VM, but inevitably came back to needing to interact with the VM too (hence the tight coupling part)

Given the "approved" PR, will consider this as optional and will defer making further changes given I don't want to incur extra risk of changes right now. However, captured in [Autofill: Consider improvements to how fragment result handling is done](https://app.asana.com/0/1204610318102050/1204610318102050/f) for re-visiting this as a follow-up.